src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunCodegenCompilationBuilder.cs

```diff
@@ -19,18 +19,21 @@ public sealed class ReadyToRunCodegenCompilationBuilder : CompilationBuilder
         // These need to provide reasonable defaults so that the user can optionally skip
         // calling the Use/Configure methods and still get something reasonable back.
         private KeyValuePair<string, string>[] _ryujitOptions = Array.Empty<KeyValuePair<string, string>>();
-        private string _inputFilePath;
+        private List<string> _inputFilePaths;
         private DependencyAnalysis.ReadyToRun.DevirtualizationManager _r2rDevirtualizationManager;
 
 
-        public ReadyToRunCodegenCompilationBuilder(CompilerTypeSystemContext context, CompilationModuleGroup group, string inputFilePath)
+        public ReadyToRunCodegenCompilationBuilder(CompilerTypeSystemContext context, CompilationModuleGroup group, List<string> inputFilePaths)
             : base(context, group, new CoreRTNameMangler(new ReadyToRunNodeMangler(), false))
         {
-            _inputFilePath = inputFilePath;
+            _inputFilePaths = inputFilePaths;
             _r2rDevirtualizationManager = new DependencyAnalysis.ReadyToRun.DevirtualizationManager(group);
 
-            EcmaModule inputModule = context.GetModuleFromPath(_inputFilePath);
-            ((ReadyToRunCompilerContext)context).InitializeAlgorithm(inputModule.MetadataReader.GetTableRowCount(TableIndex.TypeDef));
+            foreach (var inputFilePath in _inputFilePaths)
+            {
+                EcmaModule inputModule = context.GetModuleFromPath(inputFilePath);
+                ((ReadyToRunCompilerContext)context).InitializeAlgorithm(inputModule.MetadataReader.GetTableRowCount(TableIndex.TypeDef));
```

Comment:
This will not work. `InitializeAlgorithm` overwrites what the previous call to `InitializeAlgorithm` did. Context will be only correct for the last input file. It will be bogus for the rest of them.

I think for this to work, we need to get rid of the need to initalize the algorithm. I sketched the plan for that in a comment on a commit in one of Tomas's prototype branches, but that seems to now be gone, so let me rewrite it:

Updating the signature of `PrepareRuntimeSpecificStaticFieldLayout` and `FinalizeRuntimeSpecificStaticFieldLayout` to also include the `DefType` for which we're computing the layout should give you enough information (you can get the owning module of the `DefType`) so that the `Initialize` step is not needed.